app/controllers/wpcc/your_contributions_controller.rb

```diff
@@ -1,13 +1,17 @@
 require_dependency 'wpcc/engine_controller'
 
 module Wpcc
-  class YourContributionsController < EngineController
+  class YourContributionsController < Wpcc::EngineController
     protect_from_forgery
     after_action :store_eligible_salary, :store_percentages, only: :new
 
     def new
-      @contribution = Wpcc::CalculatorDelegator.delegate(
-        session[:salary].to_i, session[:contribution_preference]
+      @your_contribution = Wpcc::YourContributionGenerator.new(
+        session.to_hash.slice(
```

Comment:
@tomas-stefano it won't be confusing as the name makes it clear exactly what it is